### DISCUSSION TITLE
Problem with permissions

### DISCUSSION THREAD

**Discussion by lironsam** [filamentphp/filament#1018] [Help] [answered]
Hi,

I try to set up filament admin V2 and play with it. I made my first resources called "Users"
So far so good.
Then, what I try to do is put a link on the user name to be able to edit the profile.
However, when I do the link and clink on it, I have a screen showing 403 error.

```
  public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('name')->url(fn (User $record): string => route('filament.resources.users.edit', ['record' => $record])),
                Tables\Columns\TextColumn::make('email')->searchable()->sortable(),
                Tables\Columns\TextColumn::make('email_verified_at')
                    ->dateTime(),
                Tables\Columns\TextColumn::make('created_at')
                    ->dateTime(),
                Tables\Columns\TextColumn::make('updated_at')
                    ->dateTime(),
                Tables\Columns\TextColumn::make('avatar'),
                Tables\Columns\BooleanColumn::make('is_admin'),
                Tables\Columns\BooleanColumn::make('activate_notification'),
                Tables\Columns\TextColumn::make('thresh_shock'),
            ])
            ->filters([
                //
            ])->prependActions([
            Tables\Actions\LinkAction::make('delete')
                ->action(fn (User $user) => $user->delete())
                ->requiresConfirmation()
                ->color('danger'),
        ]);
        ;
    }
```

I noticed however, that I can access to my profile and edit without problems but for others users, I cannot do that. 
I noticed as well that I had to change the route "users.edit" to 'filament.resources.users.edit' because the route was not recognized.
Even by calling create on the address bar like "/users/1/create" I have this restriction.

Could you please help me ?

**Comment by danharrin** [comment, marked as answer]
Hey! A few things...

1) You shouldn't need that custom `url()` at all - table rows should be linked to edit the record by default if you have permission.
2) I assume that the problem you're facing is because you have a `UserPolicy` in your app that only allows users to update their own profile. Make sure admins can too.
3) (Totally unrelated but) the custom `LinkAction` has a `$user` parameter. We use dependency injection here, so the parameter needs to be called `$record` or the user won't be injected properly.

**Reply by lironsam** [reply]
Thank for your fast answer ! Indeed, I had an user policy ! I've updated and now it worked ! Thank a lot @danharrin

**Reply by danharrin** [reply]
Awesome, glad it worked!
